blog/_posts/2021-01-19-scala-native-0.4-release.md

```diff
@@ -0,0 +1,225 @@
+The last one was introduced to support Scala 2.12 behavior which compiles trait methods as default methods in Java interfaces.
+
+Although we always try to keep our NIR changes to absolute minimum, the number of removed or no longer used primitives used 
+in the previous version forced us to break backward binary compatibility. This decision allows us to remove number of known bugs
+and easier maintenance in the future. NIR format might still change in following minor releases of Scala Native, but they
+would be fully backward-compatible from now on.
+
+## How about existing libraries?
+Scala Native had many [libraries](https://index.scala-lang.org/search?targetTypes=native&scalaNativeVersions=0.4.0-M2) compatible with SN `0.4.0-M2`.
+Unfortunately, the new release is not binary compatible with them, and you'll need to wait until authors will republish them using Scala Native 0.4.0.
+
+However, with increasing interest in running the Scala code natively, it should happen quickly after the release.
+
+## Contributing
+Contributors are always welcome, you can support the Scala Native project in multiple ways.
+The most straightforward way of doing so is by working on the plugin itself, take a look at our [contributors’ guide](https://scala-native.readthedocs.io/en/latest/contrib/index.html).
+
+More importantly however, it is to make our ecosystem more native-compatible, this means developing tools and libraries that are not dependent from Java internals.
+For example, you can take a look at [sconfig](https://github.com/ekrich/sconfig) - a Scala port of the widely used Lightbend Config
+
+> Only Scala source files can be used by the Scala Native plugin.
+> Distributed libraries contain `.nir` files in their JARs which are crucial in making the SN plugin able to link them correctly.
+
+## Summary
+I think you can agree with me now, that Scala Native is starting to keep up with the rest of the Scala ecosystem.
+We are expecting it would bloom short after the release with a lot of libraries and tools.
+With introduced support for recent Scala versions it is much more approachable for the users, and combined with easier testing 
+may improve its overall quality.
+
+Even though resources allocated into this project were quite limited, we have managed to fulfill all our goals announced 
+last year in [Scala Native Next Steps](https://contributors.scala-lang.org/t/scala-native-next-steps/4216). Currently, this
+project is developed by a single engineer working full-time, thanks to our cooperation with VirtusLab. 
+We also appreciate and would like to thank our community contributors for the huge amounts of work they have done for this project.
+
+There are many plans for the future of Scala Native, few things you can expect is for sure integration with the Scala 3 
+and better support for different architectures, like ARM. Stay tuned for next updates...
+
```

Comment:
We have many plans for the future of Scala Native. Our top priorities include Scala 3 support and support for more architectures such as ARM.